src/coreComponents/codingUtilities/TableFormatter.cpp

```diff
@@ -0,0 +1,399 @@
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableFormatter.cpp
+ */
+
+#include "codingUtilities/TableFormatter.hpp"
+namespace geos
+{
+
+TableFormatter::TableFormatter( TableLayout & tableLayout )
+{
+  m_tableLayout = tableLayout;
+}
+
+void TableFormatter::fillTableColumnsFromRows( std::vector< TableLayout::Column > & columns,
+                                               std::vector< std::vector< string > > const & rows )
+{
+  for( size_t idxRow = 0; idxRow < rows.size(); idxRow++ )
+  {
+    for( size_t idxColumn = 0; idxColumn < columns.size(); idxColumn++ )
+    {
+      columns[idxColumn].columnValues.push_back( rows[idxRow][idxColumn] );
+    }
+  }
+}
+
+///////////////////////////////////////////////////////////////////////
+////// CSV Formatter implementation
+///////////////////////////////////////////////////////////////////////
+
+TableCSVFormatter::TableCSVFormatter( TableLayout & tableLayout )
+{
+  m_tableLayout = tableLayout;
+}
+
+string_view TableCSVFormatter::headerToString()
+{
+  string headerValues = "";
+  string separator = ",";
+
+  for( std::size_t idxColumn = 0; idxColumn < m_tableLayout.m_columns.size(); ++idxColumn )
+  {
+    headerValues += m_tableLayout.m_columns[idxColumn].parameter.columnName;
+    if( idxColumn < m_tableLayout.m_columns.size() - 1 )
+    {
+      headerValues += separator;
+    }
+  }
+  headerValues += "\n";
+  return headerValues;
+}
+
+string_view TableCSVFormatter::dataToString( TableData tableData )
+{
+  std::vector< std::vector< string > > rowsValues = tableData.getTableDataRows();
+  std::vector< TableLayout::Column > columns = m_tableLayout.m_columns;
+  string data;
+
+  fillTableColumnsFromRows( columns, rowsValues );
+
+  for( const auto & row : rowsValues )
+  {
+    for( size_t idxColumn = 0; idxColumn < row.size(); ++idxColumn )
+    {
+      data += row[idxColumn];
+      if( idxColumn < row.size() - 1 )
+      {
+        data += ",";
+      }
+    }
+    data += "\n";
+  }
+  return data;
+}
+
+///////////////////////////////////////////////////////////////////////
+////// Log Formatter implementation
+///////////////////////////////////////////////////////////////////////
+
+/**
+ * @brief Build a value cell given an alignment and spaces from "|"
+ *
+ * @param alignment
+ * @param value
+ * @param spaces
+ * @return A cell value
+ */
+string buildValueCell( TableLayout::Alignment const alignment, string_view value, integer const spaces )
+{
+  switch( alignment )
+  {
+    case TableLayout::right:   return GEOS_FMT( "{:>{}}", value, spaces );
+    case TableLayout::left:    return GEOS_FMT( "{:<{}}", value, spaces );
+    case TableLayout::middle:  return GEOS_FMT( "{:^{}}", value, spaces );
+    default:             return GEOS_FMT( "{:<{}}", value, spaces );
+  }
```

Comment:
I think that the [previous comment of Thomas](https://github.com/GEOS-DEV/GEOS/pull/2984#discussion_r1507986672) were suggesting that data should aligned differently by default (to the right should be a better convention, and it is the default in `ColumnParam`).
Anyway, what the returns are no longer aligned ?

Reply:
fixed